collections/_posts/2024-03-20-ImageBind.md

```diff
@@ -0,0 +1,121 @@
+
+&nbsp;
+
+# ImageBind
+
+<div style="text-align:center">
+<img src="/collections/images/imagebind/overview.jpg" width=800></div>
+<p style="text-align: center;font-style:italic">Figure 1. Overview of ImageBind.</p>
+
+ImageBind uses pairs of modalities $$(\mathcal{I}, \mathcal{M})$$ where $$\mathcal{I}$$ represents images and $$M$$ another modality. In their work, $$\mathcal{M}$$ can be text, audio, depth sensor data, thermal data, and Inertial Measurment Unit (IMU, times-series data from accelerometers and gyroscopes). Note that image can replaced by image by inflating patch projection layers, thus both are treated with the same encoder.
+
+Each modality $$M$$ has its own dedicated encoder (see below for details) and an associated projector $$f_\mathcal{M}$$, so that every modality embedding ends up in a space of common dimension.
+
+The model is trained with the standard InfoNCE contrastive loss. Given a batch of observations, it aims to make the embeddings of a pair ($$\mathcal{I}_i$$, $$\mathcal{M}_i$$) closer while pulling away the 'negative' observations, that is all the other pairs of data in the batch. The loss can be written :
+
+$$
+\mathcal{L}_{\mathcal{I},\mathcal{M}} = - \log \frac{\exp(\textbf{q}_i^T \textbf{k}_i / \tau)}{\exp(\textbf{q}_i^T \textbf{k}_i / \tau) + \sum_{i\neq j} \exp(\textbf{q}_i^T \textbf{k}_j/\tau)}
+$$
+
+&nbsp;
+
+## Modality encoders details
+
+* **Image and video :** Vision Transformer (ViT). For video , they use 2-frame clips, with a video patch of size $$2 \times 16 \times 16$$ ($$T\times H \times W$$). Encode's weights are inflated to work with spatiotemporal patches and at inference, features from successive 2-frame clips are aggregated. They leverage image-text supervision from large-scale data by using OpenCLIP's image encoder weights (ViT-H, 630M parameters). The weights are kept frozen.
+
+* **Audio :** Audio samples are recorded at 16kHz. They are cut into 2-second samples and converted into spectrograms using 128-mel spectrograms bins. Spectrograms are processed as images with ViT with a patch size of 16 and stride 10.
+
+* **Text :** They directly use CLIP encoder architecture, which is a VIT. Again, Tyey leverage image-text supervision from large-scale data by using OpenCLIP's text encoder weights (320M parameters). The weights are kept frozen.
+
+* **Depth and Thermal data :** They treat both as one-channel images and use ViT to encoder them. 
+
+* **IMU :** The 6-channel signal acquisition is cut into 5-second clips, giving 2K time steps IMU. They are projected using 1D convolution with kernel size of 8 and then encoded using a Transformer.
+
+&nbsp;
+
+# Experiments
+
+## Datasets
```

Comment:
Is the model trained once on the combination of all the datasets? If so, what is the size of the full training dataset?